### PR TITLE
feat: socials component

### LINKED ISSUE #537: Socials component
### Check for existing issues

- [ ] Completed

### Describe the feature

image

Component for the socials of the user 

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by AfonsoMartins26**

### CODE REVIEW

File: lib/atomic/uploaders/profile_picture.ex [hunk 1]
```diff
@@ -2,7 +2,7 @@ defmodule Atomic.Uploaders.ProfilePicture do
   @moduledoc """
   Uploader for profile pictures.
   """
-  use Atomic.Uploader, extensions: ~w(.jpg .jpeg .png .gif)
+  use Atomic.Uploader, extensions: ~w(.jpg .jpeg .png .gif), max_size: 500
```
Comment (MarioRodrigues10) [inline lib/atomic/uploaders/profile_picture.ex]:
```suggestion
  use Atomic.Uploader, extensions: ~w(.jpg .jpeg .png .gif)
```

File: lib/atomic_web/components/socials.ex [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  def socials(assigns) do
+    ~H"""
+    <%= if @entity.socials do %>
+      <div class="mt-4 flex gap-4">
+        <%= for {social, icon, url_base} <- [
+            {:tiktok, "tiktok.svg", "https://tiktok.com/"},
+            {:instagram, "instagram.svg", "https://instagram.com/"},
+            {:facebook, "facebook.svg", "https://facebook.com/"},
+            {:x, "x.svg", "https://x.com/"}
+          ] do %>
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]: Socials list should be configured inside the socials component, please define an outside function or variable which returns the socials.

```suggestion
      <div class="mt-4 flex gap-4">
        <%= for {social, icon, url_base} <- [
            {:tiktok, "tiktok.svg", "https://tiktok.com/"},
            {:instagram, "instagram.svg", "https://instagram.com/"},
            {:facebook, "facebook.svg", "https://facebook.com/"},
            {:x, "x.svg", "https://x.com/"}
          ] do %>
```

File: lib/atomic_web/components/socials.ex [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          <% social_value = Map.get(@entity.socials, social) %>
+          <%= if social_value do %>
+            <div class="flex flex-row items-center gap-x-1">
+              <img src={"/images/" <> icon} class="h-5 w-5" alt={social |> Atom.to_string() |> String.capitalize()} />
+              <.link class="text-blue-500" target="_blank" href={url_base <> social_value}>
+                <%= social |> Atom.to_string() |> String.capitalize() %>
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]:
```suggestion
                <%= some_function(social)  %>
```
Please use a function from helpers module for this

File: lib/atomic_web/live/partner_live/show.html.heex [hunk 4]
```diff
@@ -21,37 +21,27 @@
...
+              <.icon name="hero-map-pin" class="h-5 w-5 text-zinc-400" />
+              <.link class="text-blue-500" href={"https://www.google.com/maps/search/?api=1&query=#{@partner.location.name}"}>
+                <%= @partner.location.name %>
+              </.link>
             </div>
           <% end %>
+
           <%= if @partner.socials do %>
-            <div class="grid grid-cols-2 grid-rows-3 lg:flex lg:flex-row lg:gap-x-4">
+            <!-- Redes Sociais e Website -->
```
Comment (MarioRodrigues10) [inline lib/atomic_web/live/partner_live/show.html.heex]: Keep it in english
```suggestion
            <!-- Redes Sociais e Website -->
```

File: lib/atomic_web/live/partner_live/show.html.heex [hunk 5]
```diff
@@ -21,37 +21,27 @@
             </h1>
           </div>
           <%= if @partner.location do %>
+            <!-- Localização -->
```
Comment (MarioRodrigues10) [inline lib/atomic_web/live/partner_live/show.html.heex]: Keep it in english
```suggestion
            <!-- Localização -->
```

File: lib/atomic_web/live/partner_live/show.html.heex [hunk 6]
```diff
@@ -21,37 +21,27 @@
...
-                  <.link class="text-blue-500" href={"https://facebook.com/" <> @partner.socials.facebook}>Facebook</.link>
-                </div>
-              <% end %>
-              <%= if @partner.socials.x do %>
-                <div class="flex flex-row items-center gap-x-1 gap-y-0 sm:gap-y-2">
-                  <img src="/images/x.svg" class="size-5" alt="X" />
-                  <.link class="text-blue-500" href={"https://x.com/" <> @partner.socials.x}>X</.link>
-                </div>
-              <% end %>
+              <!-- Redes Sociais -->
```
Comment (MarioRodrigues10) [inline lib/atomic_web/live/partner_live/show.html.heex]: Keep it in english
```suggestion
              <!-- Redes Sociais -->
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

File: lib/atomic_web/components/socials.ex [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    ~H"""
+    <%= if @entity.socials do %>
+      <div class="mt-4 flex gap-4">
+        <%= for {social, icon, url_base} <- [
+            {:tiktok, "tiktok.svg", "https://tiktok.com/"},
+            {:instagram, "instagram.svg", "https://instagram.com/"},
+            {:facebook, "facebook.svg", "https://facebook.com/"},
+            {:x, "x.svg", "https://x.com/"}
+          ] do %>
+          <% social_value = Map.get(@entity.socials, social) %>
```
Comment (joaodiaslobo) [inline lib/atomic_web/components/socials.ex]: This is a known pitfall when using heex templates. Learn more about it [here](https://hexdocs.pm/phoenix_live_view/assigns-eex.html#pitfalls).

File: lib/atomic_web/components/socials.ex (hunk 2 above)
Comment (joaodiaslobo) [inline lib/atomic_web/components/socials.ex]:
```suggestion
  attr :socials, :map, required: true
  
  def socials(assigns) do
    ~H"""
      <div class="mt-4 flex gap-4">
        <%= for {social, icon, url_base} <- [
            {:tiktok, "tiktok.svg", "https://tiktok.com/"},
            {:instagram, "instagram.svg", "https://instagram.com/"},
            {:facebook, "facebook.svg", "https://facebook.com/"},
            {:x, "x.svg", "https://x.com/"}
          ] do %>
```

File: lib/atomic_web/live/partner_live/show.html.heex [hunk 8]
```diff
@@ -21,37 +21,27 @@
...
+              <.socials entity={@partner} />
```
Comment (joaodiaslobo) [inline lib/atomic_web/live/partner_live/show.html.heex]:
```suggestion
  <%= if @partner.socials do %>
              <.socials entity={@partner.socials} />
  <% end %>
```

File: lib/atomic_web/live/partner_live/show.ex [hunk 9]
```diff
@@ -2,6 +2,7 @@ defmodule AtomicWeb.PartnerLive.Show do
   use AtomicWeb, :live_view
 
   import AtomicWeb.Components.Avatar
+  import AtomicWeb.Components.Socials
```
Comment (joaodiaslobo) [inline lib/atomic_web/live/partner_live/show.ex]:
```suggestion
  import AtomicWeb.Components.{Avatar, Socials}
```

**Review by joaodiaslobo — DISMISSED** [review]

File: lib/atomic_web/components/socials.ex [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  use Phoenix.Component
+
+  attr :socials, :map, required: true
+  attr :entity, :map, required: true
+
+  def socials(assigns) do
+    ~H"""
+    <div class="mt-4 flex gap-4">
+      <%= for {social, icon, url_base} <- get_socials() do %>
+        <% social_value = Map.get(@entity, social) %>
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]: This can't be defined here
```suggestion
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

File: lib/atomic_web/components/socials.ex [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  use Phoenix.Component
+  alias AtomicWeb.Components.Helpers
+
+  attr :entity, :map, required: true
+
+  def socials(assigns) do
+    assigns = assign(assigns, :socials_with_values, get_social_values(assigns.entity))
+
+    ~H"""
+    <div class="mt-4 flex gap-4">
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]:
```suggestion
    <div class="flex gap-4">
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

File: lib/atomic_web/components/socials.ex [hunk 12]
```diff
@@ -0,0 +1,43 @@
+defmodule AtomicWeb.Components.Socials do
+  @moduledoc false
+  use Phoenix.Component
+  alias AtomicWeb.Components.Helpers
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]:
```suggestion
  use Atomic.Web, :component
```

File: lib/atomic_web/components/helpers.ex [hunk 13]
```diff
@@ -101,4 +101,10 @@ defmodule AtomicWeb.Components.Helpers do
     |> maybe_put(:order_by, order_by)
     |> maybe_put(:order_directions, order_directions)
   end
+
+  def atom_to_string_capitalize(atom) when is_atom(atom) do
+    atom
+    |> Atom.to_string()
+    |> String.capitalize()
+  end
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/helpers.ex]: This belongs to `helpers.ex` module not `helpers.ex` component

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

**Review by FilipeR13 — COMMENTED** [review]
After Mario suggestions, it may be ready to merge

File: lib/atomic_web/components/socials.ex [hunk 14]
```diff
@@ -0,0 +1,45 @@
+defmodule AtomicWeb.Components.Socials do
+  @moduledoc false
+
+  use Atomic.Web, :component
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]:
```suggestion
  use AtomicWeb, :component
```

File: lib/atomic_web/components/socials.ex [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+
+  alias AtomicWeb.Helpers
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]:
```suggestion
```

File: lib/atomic_web/components/socials.ex [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    assigns = assign(assigns, :socials_with_values, get_social_values(assigns.entity))
+
+    ~H"""
+    <div class="flex gap-4">
+      <%= for {social, icon, url_base, social_value} <- @socials_with_values do %>
+        <%= if social_value do %>
+          <div class="flex flex-row items-center gap-x-1">
+            <img src={"/images/" <> icon} class="h-5 w-5" alt={Helpers.atom_to_string_capitalize(social)} />
+            <.link class="text-blue-500" target="_blank" href={url_base <> social_value}>
+              <%= Helpers.atom_to_string_capitalize(social) %>
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]:
```suggestion
            <img src={"/images/" <> icon} class="h-5 w-5" alt={atom_to_string_capitalize(social)} />
            <.link class="text-blue-500" target="_blank" href={url_base <> social_value}>
              <% atom_to_string_capitalize(social) %>
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

File: lib/atomic_web/views/helpers.ex [hunk 17]
```diff
@@ -391,4 +391,10 @@ defmodule AtomicWeb.Helpers do
   def error_to_string(:too_large), do: gettext("Too large")
   def error_to_string(:not_accepted), do: gettext("You have selected an unacceptable file type")
   def error_to_string(:too_many_files), do: gettext("You have selected too many files")
+
+  def atom_to_string_capitalize(atom) when is_atom(atom) do
+    atom
+    |> Atom.to_string()
+    |> String.capitalize()
+  end
```
Comment (MarioRodrigues10) [inline lib/atomic_web/views/helpers.ex]:
```suggestion
```

File: lib/atomic_web/components/socials.ex [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+    assigns = assign(assigns, :socials_with_values, get_social_values(assigns.entity))
+
+    ~H"""
+    <div class="flex gap-4">
+      <%= for {social, icon, url_base, social_value} <- @socials_with_values do %>
+        <%= if social_value do %>
+          <div class="flex flex-row items-center gap-x-1">
+            <img src={"/images/" <> icon} class="h-5 w-5" alt={atom_to_string_capitalize(social)} />
+            <.link class="text-blue-500" target="_blank" href={url_base <> social_value}>
+              <%= atom_to_string_capitalize(social) %>
```
Comment (MarioRodrigues10) [inline lib/atomic_web/components/socials.ex]:
```suggestion
            <img src={"/images/" <> icon} class="h-5 w-5 capitalize" alt={Atom.to_string(social)} />
            <.link class="text-blue-500 capitalize" target="_blank" href={url_base <> social_value}>
              <%= Atom.to_string(social) %>
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

**Review by MarioRodrigues10 — APPROVED** [review]
:rocket:

**Pull request merged**